src/screens/Search/SearchScreen/index.js

```diff
@@ -30,55 +30,43 @@ class SearchScreen extends Component {
       manga: [],
       users: [],
     },
-    index: 0,
-    routes: [
-      {
-        key: 'anime',
-        title: 'Anime',
+    scenes: {
+      anime: {
         apiKey: this.props.algoliaKeys.media && this.props.algoliaKeys.media.key,
         indexName: this.props.algoliaKeys.media && this.props.algoliaKeys.media.index,
         kind: 'anime',
       },
-      {
-        key: 'manga',
-        title: 'Manga',
+      manga: {
         apiKey: this.props.algoliaKeys.media && this.props.algoliaKeys.media.key,
         indexName: this.props.algoliaKeys.media && this.props.algoliaKeys.media.index,
         kind: 'manga',
       },
-      {
-        key: 'users',
-        title: 'Users',
+      users: {
         apiKey: this.props.algoliaKeys.users && this.props.algoliaKeys.users.key,
         indexName: this.props.algoliaKeys.users && this.props.algoliaKeys.users.index,
       },
-    ],
+    }
   };
 
-  doSearch = (query, route) => {
-    if (!route || !route.apiKey) return;
-
-    const algoliaClient = algolia(kitsuConfig.algoliaAppId, route.apiKey);
-    const algoliaIndex = algoliaClient.initIndex(route.indexName);
-
-    const filters = route.kind ? `kind:${route.kind}` : '';
+  handleQuery = (scene, query) => {
+   const nextState = { ...this.state.query, [scene]: isEmpty(query) ? undefined : query };
+   this.setState({ query: nextState }, () => {
+     this.debouncedSearch(query, scene);
+   });
+  };
 
-    algoliaIndex.search({ query, filters }, (err, content) => {
+  executeSearch = (query, scene) => {
+    const currentScene = this.state.scenes[scene];
+    const client = algolia(kitsuConfig.algoliaAppId, currentScene.apiKey);
```

Comment:
There could be a case here where `currentScene.apiKey` is null, and if so algolia won't be initialized, maybe just return early?

Reply:
Yeah, that sucks :(

Reply:
Fixed, we should maybe flip a flag there and show a failed state.